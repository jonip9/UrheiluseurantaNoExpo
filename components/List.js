import 'react-native-gesture-handler';
import React, {useState, useEffect} from 'react';
import {Text, View, Button, Modal, FlatList} from 'react-native';
import {useIsFocused} from '@react-navigation/native';
import {styles} from '../styles/styles';

export default function List({route, navigation}) {
  const [data, setData] = useState([]);
  const [error, setError] = useState('');
  const [refresh, setRefresh] = useState(false);
  const isFocused = useIsFocused();
  const {id} = route.params;
  const {user} = route.params;
  const url = 'http://192.168.1.102:3000/event/all/' + id;

  useEffect(() => {
    if (isFocused) {
      fetchAllEvents();
    }
  }, [isFocused]);

  async function fetchAllEvents() {
    fetch(url, {
      method: 'GET',
    })
      .then((response) => {
        if (!response.ok) {
          throw new Error('Error');
        }
        return response.json();
      })
      .then((data) => {
        setData(data);
        setRefresh(false);
        console.log('Success: ', data);
      })
      .catch((error) => {
        setError(error);
        setRefresh(false);
        console.error('Error: ', error);
      });
  }

  function handleRefresh() {
    setRefresh(true);
    fetchAllEvents();
  }

  return (
    <View style={styles.container}>
      <FlatList
        data={data}
        renderItem={({item}) => (
          <ListItem details={item} nav={navigation} refresh={fetchAllEvents} />
        )}
        keyExtractor={(item) => JSON.stringify(item.id)}
        onRefresh={() => handleRefresh()}
        refreshing={refresh}
      />
      <View style={{marginTop: 10}}>
        <Button
          title="Lisää"
          onPress={() =>
            navigation.navigate('Tiedot', {userId: id, itemData: null})
          }
        />
      </View>
    </View>
  );
}

function ListItem({details, nav, refresh}) {
  const [modalVisible, setModalVisible] = useState(false);
  const url = 'http://192.168.1.102:3000/event/delete';

  async function deleteEvent() {
    fetch(url, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({id: details.id}),
    })
      .then((response) => {
        if (!response.ok) {
          throw new Error('Error');
        }
        return response.json();
      })
      .then((data) => {
        setModalVisible(false);
        refresh();
        console.log('Success: ', data);
      })
      .catch((error) => {
        setModalVisible(false);
        console.error('Error: ', error);
      });
  }

  return (
    <View style={styles.container}>
      <Modal transparent={true} visible={modalVisible} style={styles.modal}>
        <View style={styles.centerModal}>
          <View style={styles.modal}>
            <Text>Haluatko varmasti poistaa tapahtuman?</Text>
            <View style={styles.buttonModal}>
              <Button title="Kyllä" onPress={() => deleteEvent()} />
            </View>
            <View style={styles.buttonModal}>
              <Button
                title="Ei"
                onPress={() => setModalVisible(!modalVisible)}
              />
            </View>
          </View>
        </View>
      </Modal>

      <View>
        <Text>{details.date.slice(0, 10)}</Text>
        <Text>{details.type}</Text>
      </View>
      <View>
        <Text>Aika: {details.duration} h</Text>
        <Text>Matka: {details.distance} km</Text>
        <Text>{details.comment}</Text>
      </View>
      <View style={styles.button}>
        <Button title="Poista" onPress={() => setModalVisible(true)} />
        <Button
          title="Muokkaa"
          onPress={() =>
            nav.navigate('Tiedot', {userId: details.user, itemData: details})
          }
        />
      </View>
    </View>
  );
}
