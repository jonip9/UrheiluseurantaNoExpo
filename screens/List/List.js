import 'react-native-gesture-handler';
import React, { useState, useEffect } from 'react';
import { View, Button, FlatList } from 'react-native';
import { useIsFocused } from '@react-navigation/native';
import ListItem from '../../components/ListItem';
import { styles } from '../../styles/styles';

export default function List({ route, navigation }) {
  const [data, setData] = useState([]);
  const [error, setError] = useState('');
  const [refresh, setRefresh] = useState(false);
  const isFocused = useIsFocused();
  //const { id } = route.params;
  //const { user } = route.params;
  //const url = 'http://192.168.1.102:3000/event/all/' + id;

  useEffect(() => {
    if (isFocused) {
      //fetchAllEvents();
    }
  }, [isFocused]);

  async function fetchAllEvents() {
    fetch('url', {
      method: 'GET',
    })
      .then((response) => {
        if (!response.ok) {
          throw new Error('Error');
        }
        return response.json();
      })
      .then((data) => {
        setData(data);
        setRefresh(false);
        console.log('Success: ', data);
      })
      .catch((error) => {
        setError(error);
        setRefresh(false);
        console.error('Error: ', error);
      });
  }

  function handleRefresh() {
    setRefresh(true);
    fetchAllEvents();
  }

  return (
    <View style={styles.container}>
      <FlatList
        data={data}
        renderItem={({ item }) => (
          <ListItem details={item} nav={navigation} refresh={fetchAllEvents} />
        )}
        keyExtractor={(item) => JSON.stringify(item.id)}
        onRefresh={() => handleRefresh()}
        refreshing={refresh}
      />
      <View style={{ marginTop: 10 }}>
        <Button
          title="Lisää"
          onPress={() =>
            navigation.navigate('Tiedot', { userId: 'id', itemData: null })
          }
        />
      </View>
    </View>
  );
}
