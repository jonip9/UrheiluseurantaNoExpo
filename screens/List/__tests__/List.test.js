import 'react-native';
import React from 'react';
import List from '../List';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { NavigationContainer } from '@react-navigation/native';

import renderer, { act } from 'react-test-renderer';

test('App renders correctly', async () => {
  jest.useFakeTimers();
  const { Screen, Navigator } = createBottomTabNavigator();

  const tree = renderer
    .create(
      <NavigationContainer>
        <Navigator>
          <Screen name="List" component={List} initialParams={{ id: 0 }} />
        </Navigator>
      </NavigationContainer>,
    )
    .toJSON();
  await act(async () => {});
  expect(tree).toMatchSnapshot();
});
