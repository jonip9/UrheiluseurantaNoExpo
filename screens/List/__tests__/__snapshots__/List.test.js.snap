// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={false}
        pointerEvents="auto"
        removeClippedSubviews={false}
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            null,
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "margin": 10,
                  "padding": 10,
                }
              }
            >
              <RCTScrollView
                data={Array []}
                disableVirtualization={false}
                getItem={[Function]}
                getItemCount={[Function]}
                horizontal={false}
                initialNumToRender={10}
                keyExtractor={[Function]}
                maxToRenderPerBatch={10}
                onContentSizeChange={[Function]}
                onEndReachedThreshold={2}
                onLayout={[Function]}
                onMomentumScrollEnd={[Function]}
                onRefresh={[Function]}
                onScroll={[Function]}
                onScrollBeginDrag={[Function]}
                onScrollEndDrag={[Function]}
                refreshControl={
                  <RefreshControlMock
                    onRefresh={[Function]}
                    refreshing={false}
                  />
                }
                refreshing={false}
                removeClippedSubviews={false}
                renderItem={[Function]}
                scrollEventThrottle={50}
                stickyHeaderIndices={Array []}
                updateCellsBatchingPeriod={50}
                viewabilityConfigCallbackPairs={Array []}
                windowSize={21}
              >
                <RCTRefreshControl />
                <View />
              </RCTScrollView>
              <View
                style={
                  Object {
                    "marginTop": 10,
                  }
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={Object {}}
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {},
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#007AFF",
                            "fontSize": 18,
                            "margin": 8,
                            "textAlign": "center",
                          },
                        ]
                      }
                    >
                      Lisää
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="auto"
      style={
        Object {
          "backgroundColor": "rgb(255, 255, 255)",
          "borderTopColor": "rgb(216, 216, 216)",
          "borderTopWidth": 0.5,
          "bottom": 0,
          "elevation": 8,
          "height": 49,
          "left": 0,
          "paddingBottom": 0,
          "paddingHorizontal": 0,
          "position": null,
          "right": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityLabel="List, tab, 1 of 1"
          accessibilityRole="button"
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "textAlign": "center",
                },
                Object {
                  "color": "rgb(0, 122, 255)",
                },
                Object {
                  "fontSize": 10,
                },
                undefined,
              ]
            }
          >
            List
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
